src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy

```diff
@@ -49,8 +50,20 @@ class IqPolicyEvaluatorUtil
           new IqClientFactoryConfiguration(credentialsId: iqPolicyEvaluator.jobCredentialsId, context: run.parent,
               log: loggerBridge))
 
-      def verified = iqClient.verifyOrCreateApplication(applicationId)
-      checkArgument(verified, 'The application ID ' + applicationId + ' is invalid.')
+      try {
```

Comment:
> I do wonder if it would be cleaner at this point to extract all this to its own method

I could move all the logic into another method.  Return void and the next call after calling this method could check the run.status.  if it is bad, do a return null.  I see that being done in other parts of the code.

Example: 

```
      verifyOrCreateApplication(iqClient, listener, run, applicationId)
      if (run.getResult() == Result.FAILURE) {
        return null
      }
```
The method:

```
private static void verifyOrCreateApplication(final InternalIqClient iqClient, final TaskListener listener, final Run run, final String applicationId) {
    try {
      def verified = iqClient.verifyOrCreateApplication(applicationId)
      checkArgument(verified, 'The application ID ' + applicationId + ' is invalid.')
    }
    catch (IqClientException e) {
      if (isHttp404Error(e)) {
        listener.logger.println Messages._IqPolicyEvaluation_AutoAppMissing()
        run.result = Result.FAILURE
      }
      else {
        throw e
      }
    }
  }
  ```